hazelcast-client/src/main/java/com/hazelcast/client/impl/statistics/Statistics.java

```diff
@@ -0,0 +1,254 @@
+            stats.put(buffer.append(pathPrefix).append("misses").toString(), Long.toString(nearCacheStats.getMisses()));
+            buffer.setLength(0);
+            stats.put(buffer.append(pathPrefix).append("ownedEntryCount").toString(),
+                    Long.toString(nearCacheStats.getOwnedEntryCount()));
+            buffer.setLength(0);
+            stats.put(buffer.append(pathPrefix).append("expirations").toString(), Long.toString(nearCacheStats.getExpirations()));
+            buffer.setLength(0);
+            stats.put(buffer.append(pathPrefix).append("ownedEntryMemoryCost").toString(),
+                    Long.toString(nearCacheStats.getOwnedEntryMemoryCost()));
+            buffer.setLength(0);
+            stats.put(buffer.append(pathPrefix).append("lastPersistenceFailure").toString(),
+                    nearCacheStats.getLastPersistenceFailure());
+        }
+    }
+
+    private void sendStats(Map<String, String> newStats, ClientConnection ownerConnection) {
+        ClientMessage request = ClientStatisticsCodec.encodeRequest(newStats.entrySet());
+        try {
+            new ClientInvocation(client, request, ownerConnection).invoke();
+        } catch (Exception e) {
+            // suppress exception, do not print too many messages
+            if (logger.isFinestEnabled()) {
+                logger.finest("Could not send stats ", e);
+            }
+        }
+    }
+
+    private void registerStatistics() {
+        // Note that the OperatingSystemMetricSet and RuntimeMetricSet are already registered during client start,
+        // hence we do not re-register
+
+        periodicStats = new PeriodicStatistics(metricsRegistry);
+
+        staticStats.put("enterprise", Boolean.toString(enterprise));
+    }
+
+    class PeriodicStatistics {
+        private final String[] STATISTIC_NAMES = {
+                "os.committedVirtualMemorySize",
+                "os.freePhysicalMemorySize",
+                "os.freeSwapSpaceSize",
+                "os.maxFileDescriptorCount",
+                "os.openFileDescriptorCount",
+                "os.processCpuTime",
+                "os.systemLoadAverage",
+                "os.totalPhysicalMemorySize",
+                "os.totalSwapSpaceSize",
+                "runtime.availableProcessors",
+                "runtime.freeMemory",
+                "runtime.maxMemory",
+                "runtime.totalMemory",
+                "runtime.uptime",
+                "runtime.usedMemory",
+                "executionService.userExecutorQueueSize"
+        };
+
+        private final Map<String, StringGauge> allMetrics = new HashMap<String, StringGauge>(STATISTIC_NAMES.length);
+
+        PeriodicStatistics(final MetricsRegistry metricsRegistry) {
+            for (String name : STATISTIC_NAMES) {
+                allMetrics.put(name, metricsRegistry.newStringGauge(name));
+            }
+        }
+
+        void fillMetrics(final Map<String, String> metricsMap) {
```

Comment:
I'm questioning the type of the 'metricsMap'.

If you would just send a formatted String to the other side, then only a single String is needed.